### ISSUE TITLE
ex6 - make it modular

### ISSUE THREAD

**Issue by Mark-McCracken** [nodeschool/discussions#1858]
Hi guys, so I have my code behaving exactly as it should when I run it to try it out, but the test does not pass. After checking the code it's comparing agains, and making them more similar, still didn't pass. So I just straight up copied both the files from /usr/local/lib/node_modules/learnyounode/exercises/make_it_modular/solutions
and ran them through the verify, which still failed with the following output. If it's failing against itself, I don't think there's much hope for me ever solving this.

Anybody know of a way to mark this as complete so I can move on to the next item? Or is there actually something wrong with the code? or a bug in the package?

Any help appreciated.
- node version: 4.5.0
- npm version:  2.15.9
### Error output

```
There was an error: { [Error: ENOTDIR: not a directory, scandir '/var/folders/n1/5j8vdpwn7pn9td85hvclvx4h0000gn/T/_learnyounode_18309.txt']
  errno: -20,
  code: 'ENOTDIR',
  syscall: 'scandir',
  path: '/var/folders/n1/5j8vdpwn7pn9td85hvclvx4h0000gn/T/_learnyounode_18309.txt' }

Your submission results compared to the expected:

                 ACTUAL                                 EXPECTED                
────────────────────────────────────────────────────────────────────────────────

   ""                                  !=    "46"                               

────────────────────────────────────────────────────────────────────────────────

  ✗  Submission results did not match expected!  

```
### My Code

```
file
var filterFn = require('./solution_filter.js')
var dir = process.argv[2]
var filterStr = process.argv[3]

filterFn(dir, filterStr, function (err, list) {
  if (err)
    return console.error('There was an error:', err)

  list.forEach(function (file) {
    console.log(file)
  })
})

```

module
var fs = require('fs')
var path = require('path')

module.exports = function (dir, filterStr, callback) {

  fs.readdir(dir, function (err, list) {
    if (err)
      return callback(err)

```
list = list.filter(function (file) {
  return path.extname(file) === '.' + filterStr
})

callback(null, list)
```

  })
}

```
```

**Comment by alexpetroni** [comment]
try upgrade to node 6.7.0

**Comment by webgast** [comment]
Did that:

nvm install 6.7.0
nvm use 6.7.0

(reinstall learnyounode)
npm install learnyounode

(some warnings)

`npm WARN prefer global marked@0.3.6 should be installed with -g
npm WARN prefer global learnyounode@3.5.8 should be installed with -g
chat-example@0.0.0 /home/ubuntu/workspace
└─┬ learnyounode@3.5.8 
  ├── after@0.8.2 
  ├─┬ bl@1.1.2 
  │ └─┬ readable-stream@2.0.6 
  │   ├── core-util-is@1.0.2 
  │   ├── isarray@1.0.0 
  │   ├── process-nextick-args@1.0.7 
  │   ├── string_decoder@0.10.31 
  │   └── util-deprecate@1.0.2 
  ├── boganipsum@0.1.0 
  ├─┬ colors-tmpl@1.0.0 
  │ └── colors@1.0.3 
  ├─┬ combined-stream@1.0.5 
  │ └── delayed-stream@1.0.0 
  ├─┬ concat-stream@1.5.2 
  │ ├── inherits@2.0.3 
  │ └── typedarray@0.0.6 
  ├── duplexer@0.1.1 
  ├─┬ hyperquest@2.1.1 
  │ ├─┬ duplexer2@0.0.2 
  │ │ └─┬ readable-stream@1.1.14 
  │ │   └── isarray@0.0.1 
  │ └─┬ through2@0.6.5 
  │   └─┬ readable-stream@1.0.34 
  │     └── isarray@0.0.1 
  ├─┬ rimraf@2.5.4 
  │ └─┬ glob@7.1.1 
  │   ├── fs.realpath@1.0.0 
  │   ├─┬ inflight@1.0.6 
  │   │ └── wrappy@1.0.2 
  │   ├─┬ minimatch@3.0.3 
  │   │ └─┬ brace-expansion@1.1.6 
  │   │   ├── balanced-match@0.4.2 
  │   │   └── concat-map@0.0.1 
  │   ├── once@1.4.0 
  │   └── path-is-absolute@1.0.1 
  ├── through@2.3.8 
  ├─┬ through2@2.0.1 
  │ └── xtend@4.0.1 
  ├── through2-map@3.0.0 
  ├─┬ workshopper-adventure@5.1.6 
  │ ├─┬ chalk@1.1.3 
  │ │ ├── ansi-styles@2.2.1 
  │ │ ├── escape-string-regexp@1.0.5 
  │ │ ├── has-ansi@2.0.0 
  │ │ ├── strip-ansi@3.0.1 
  │ │ └── supports-color@2.0.0 
  │ ├── combined-stream-wait-for-it@1.1.0 
  │ ├─┬ commandico@2.0.2 
  │ │ ├── explicit@0.1.1 
  │ │ ├─┬ joi@6.10.1 
  │ │ │ ├── hoek@2.16.3 
  │ │ │ ├── isemail@1.2.0 
  │ │ │ ├── moment@2.17.0 
  │ │ │ └── topo@1.1.0 
  │ │ └── minimist@1.2.0 
  │ ├─┬ i18n-core@2.1.1 
  │ │ ├── mustache@0.8.2 
  │ │ └── sprintf@0.1.5 
  │ ├─┬ latest-version@2.0.0 
  │ │ └─┬ package-json@2.4.0 
  │ │   ├─┬ got@5.7.1 
  │ │   │ ├─┬ create-error-class@3.0.2 
  │ │   │ │ └── capture-stack-trace@1.0.0 
  │ │   │ ├── duplexer2@0.1.4 
  │ │   │ ├── is-redirect@1.0.0 
  │ │   │ ├── is-retry-allowed@1.1.0 
  │ │   │ ├── is-stream@1.1.0 
  │ │   │ ├── lowercase-keys@1.0.0 
  │ │   │ ├── node-status-codes@1.0.0 
  │ │   │ ├── object-assign@4.1.0 
  │ │   │ ├─┬ parse-json@2.2.0 
  │ │   │ │ └─┬ error-ex@1.3.0 
  │ │   │ │   └── is-arrayish@0.2.1 
  │ │   │ ├─┬ pinkie-promise@2.0.1 
  │ │   │ │ └── pinkie@2.0.4 
  │ │   │ ├── read-all-stream@3.1.0 
  │ │   │ ├── timed-out@3.0.0 
  │ │   │ ├── unzip-response@1.0.2 
  │ │   │ └─┬ url-parse-lax@1.0.0 
  │ │   │   └── prepend-http@1.0.4 
  │ │   ├─┬ registry-auth-token@3.1.0 
  │ │   │ └─┬ rc@1.1.6 
  │ │   │   ├── deep-extend@0.4.1 
  │ │   │   ├── ini@1.3.4 
  │ │   │   └── strip-json-comments@1.0.4 
  │ │   ├── registry-url@3.1.0 
  │ │   └── semver@5.3.0 
  │ ├─┬ msee@0.3.2 
  │ │ ├── ansi-regex@2.0.0 
  │ │ ├── ansicolors@0.3.2 
  │ │ ├─┬ cardinal@0.7.1 
  │ │ │ ├── ansicolors@0.2.1 
  │ │ │ └─┬ redeyed@0.6.0 
  │ │ │   └── esprima@2.7.3 
  │ │ ├── marked@0.3.6 
  │ │ ├─┬ nopt@3.0.6 
  │ │ │ └── abbrev@1.0.9 
  │ │ ├─┬ table-header@0.2.2 
  │ │ │ └── repeat-string@1.6.1 
  │ │ └── text-table@0.2.0 
  │ ├─┬ simple-terminal-menu@1.1.3 
  │ │ └─┬ extended-terminal-menu@2.1.4 
  │ │   ├── charm_inheritance-fix@1.0.1 
  │ │   ├── resumer@0.0.0 
  │ │   └─┬ through2@0.6.5 
  │ │     └─┬ readable-stream@1.0.34 
  │ │       └── isarray@0.0.1 
  │ ├── split@1.0.0 
  │ ├─┬ string-to-stream@1.1.0 
  │ │ └─┬ readable-stream@2.2.2 
  │ │   └── buffer-shims@1.0.0 
  │ └─┬ workshopper-adventure-storage@3.0.0 
  │   └─┬ mkdirp@0.5.1 
  │     └── minimist@0.0.8 
  ├─┬ workshopper-exercise@2.7.0 
  │ ├── tuple-stream@0.0.2 
  │ └─┬ wcstring@2.1.1 
  │   ├── varsize-string@2.2.2 
  │   └── wcsize@1.0.0 
  └─┬ workshopper-wrappedexec@0.1.2 
    └─┬ xtend@2.1.2 
      └── object-keys@0.4.0 

npm WARN chat-example@0.0.0 No repository field.
npm WARN chat-example@0.0.0 No license field.`

and then, tried learnyounode verify modular.js again with basically the same output:

(the rest was fine)

 ✗ 

 Your additional module file [my_module.js] returned an error on its
 callback: []
